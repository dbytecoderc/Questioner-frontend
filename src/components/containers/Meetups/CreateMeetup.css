.form-control{
  height: 100vh;
}
