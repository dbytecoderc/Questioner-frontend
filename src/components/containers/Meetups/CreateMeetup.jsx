import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';

import { createMeetup } from '../../../store/actions/meetups';

import Button from '../../presentationals/Button/Button';
import Input from '../../presentationals/Input/Input';

// import Loader from '../../presentationals/Loader/Loader';

export class CreateMeetup extends Component {
  state = {
    meetupData: {
      topic: '',
      location: '',
      time: '',
    },
  };

  onChange = event => {
    this.setState({
      meetupData: {
        ...this.state.meetupData,
        [event.target.name]: event.target.value,
      },
    });
  };

  submitHandler = () => {
    const {
      history,
      // location: { state },
    } = this.props;
    this.props.createMeetup(history, this.state.meetupData);
  };
  render() {
    return (
      <div
        className="container"
        style={{
          padding: '5% 10%',
        }}
      >
        <h1
          style={{
            fontSize: '2rem',
            marginBottom: '2rem',
            textAlign: 'center',
          }}
        >
          Create a meetup
        </h1>
        <Input
          type="text"
          className="form-control form-control-lg form-rounded"
          placeholder="Meetup topic..."
          onChange={this.onChange}
          value={this.state.meetupData.topic}
          name="topic"
        />
        <br />
        <Input
          type="text"
          className="form-control form-control-lg  form-rounded"
          placeholder="Meetup address..."
          onChange={this.onChange}
          value={this.state.meetupData.location}
          name="location"
        />
        <br />
        <Input
          type="datetime-local"
          className="form-control form-control-lg  form-rounded"
          placeholder="Meetup time"
          onChange={this.onChange}
          value={this.state.meetupData.time}
          name="time"
        />
        <br />
        <Button
          className="btn-dark btn-lg submit--button"
          type="submit"
          value="Post Meetup"
          onClick={this.submitHandler}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({ meetups: state.meetups });

export default connect(
  mapStateToProps,
  { createMeetup },
)(withRouter(CreateMeetup));
