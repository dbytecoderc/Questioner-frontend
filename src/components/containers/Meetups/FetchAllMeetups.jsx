import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import fontawesome from '@fortawesome/fontawesome';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';

fontawesome.library.add(faChevronRight);

//Reducer
import { fetchMeetups } from '../../../store/actions/meetups';

import '../../../pages/AllMeetups/AllMeetups.css';

export class FetchAllMeetups extends Component {
  componentDidMount() {
    this.props.fetchMeetups();
  }
  goToMeetup = id => {
    const { history } = this.props;
    history.push(`/meetup=${id}`);
  };
  render() {
    const { meetups, isLoading } = this.props;
    let meetuplist;
    if (meetups.successResponse.data) {
      meetuplist = meetups.successResponse.data.map((meetup, key) => {
        return (
          <Fragment key={meetup.id}>
            <div
              className="card text-center mb-3  border-0"
              style={{ maxWidth: '18rem', minWidth: '18rem' }}
            >
              <div>
                <div className="img-card" href="#">
                  <img src={meetup.image_url} />
                </div>
                <div className="card-body">
                  <h4 className="card-title">{meetup.topic}</h4>
                  <p className="meetup-location">{meetup.location}</p>
                </div>
              </div>

              <div>
                <button
                  type="button"
                  className="btn btn-labeled btn-outline-dark"
                  onClick={() => {
                    this.goToMeetup(meetup.id);
                  }}
                >
                  <i className="glyphicon glyphicon-chevron-right">Attend</i>
                  <span className="right-btn-label">
                    <FontAwesomeIcon
                      className="chevron-icon"
                      icon={faChevronRight}
                    />
                  </span>
                </button>
              </div>
            </div>
          </Fragment>
        );
      });
    }

    return <>{meetuplist}</>;
  }
}

FetchAllMeetups.prototypes = {
  fetchMeetups: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ meetups: state.meetups });

export default connect(
  mapStateToProps,
  { fetchMeetups },
)(withRouter(FetchAllMeetups));
