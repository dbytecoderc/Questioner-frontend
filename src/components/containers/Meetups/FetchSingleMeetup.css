
.img-thumbnail{
  margin: 0;
}

.topic{
  font-size: 1.5rem;
  font-weight: 900;
}

.icon{
  font-size: 1.5rem;
}

.location{
  font-size: 1.2rem;
}


.meetup-container{
  min-height: 100vh;
}

.question-card{
  border-radius: 1px;
}


.thumb-btn-label {
  position: relative;
  left: -12px;
  display: inline-block;
  padding: 2px 8px;
  background: rgba(0,0,0,0.15);border-radius: 3px 0 0 3px;
  border-radius: 1px;
}

.btn-labeled {
  padding-top: 0;
  padding-bottom: 0;
}
.text-areax{
  resize: none;
}

.question-btn{
  color: #ffffff !important;
  cursor: pointer;
}
