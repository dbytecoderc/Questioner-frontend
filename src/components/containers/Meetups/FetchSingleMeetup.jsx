import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import fontawesome from '@fortawesome/fontawesome';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faMapMarkerAlt,
  faCalendarAlt,
  faThumbsUp,
  faThumbsDown,
} from '@fortawesome/free-solid-svg-icons';

import './FetchSingleMeetup.css';

fontawesome.library.add(
  faMapMarkerAlt,
  faCalendarAlt,
  faThumbsUp,
  faThumbsDown,
);

import FullPageLoader from '../../presentationals/FullPageLoader/FullPageLoader';

//Reducer
import { fetchOneMeetup } from '../../../store/actions/meetups';
import Questions from '../../presentationals/Questions/Questions';

export class FetchSingleMeetup extends Component {
  componentDidMount() {
    const {
      match: {
        params: { meetupId },
      },
      history,
    } = this.props;

    this.props.fetchOneMeetup(meetupId, history);
  }

  render() {
    const meetup = this.props.meetups.successResponse;
    const {
      meetups: { isLoading },
    } = this.props;

    let meetupQuestions;
    if (meetup.questions) {
      meetupQuestions = meetup.questions.map(question => (
        <Questions question={question} key={question.id} />
      ));
    }

    return (
      <Fragment>
        {isLoading && <FullPageLoader />}
        <div className="meetup-container">
          <div className="container ">
            <div className="row justify-content-center py-3">
              <div className="col-md-6 text-center">
                <img
                  src={meetup.image_url}
                  className="w-100 img-thumbnail"
                  alt="Cinque Terre"
                />
              </div>
              <div className="col-md-6 text-left">
                <div className="text-center">
                  <p className="font-weight-bold topic">{meetup.topic}</p>
                </div>
                <div className="mt-4">
                  <FontAwesomeIcon
                    icon={faMapMarkerAlt}
                    className="mr-4 icon"
                  />
                  <span className="location">{meetup.location}</span>
                </div>

                <div className="mt-4">
                  <FontAwesomeIcon icon={faCalendarAlt} className="mr-4 icon" />
                  <span className="location">{meetup.happening_on}</span>
                </div>
              </div>
            </div>
            <div className="questions">
              <div className="row">
                <div className="col-md-8">
                  <div className="form-group purple-border">
                    <textarea
                      placeholder="Ask a question..."
                      className="form-control text-areax"
                      id="exampleFormControlTextarea4"
                      rows="3"
                    />
                    <button
                      type="button"
                      className="btn btn-dark mt-2 text-right"
                    >
                      Post Question
                    </button>
                  </div>
                  {meetupQuestions}
                </div>
                <div className="col-md-4">
                  <div className="card">
                    <img
                      src="https://devfest.gdgla.org/assets/img/devfest.png"
                      className="card-img-top"
                      alt="..."
                    />
                    <div className="card-body">
                      <p className="card-text">
                        Some quick example text to build on the card title and
                        make up the bulk of the card's content.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

FetchSingleMeetup.prototypes = {
  fetchOneMeetup: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ meetups: state.meetups });

export default connect(
  mapStateToProps,
  { fetchOneMeetup },
)(withRouter(FetchSingleMeetup));
