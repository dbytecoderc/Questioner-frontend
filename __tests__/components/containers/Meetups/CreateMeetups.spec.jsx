import React from 'react';
import { shallow } from 'enzyme';

import { CreateMeetup } from '../../../../src/components/containers/Meetups/CreateMeetup';

const props = {
  auth: {
    isLoading: true,
    successResponse: {
      status: 'success',
      message: '',
    },
    errorResponse: [],
  },
  loginUser: jest.fn(),
};

const component = shallow(<CreateMeetup {...props} />);

describe('Login component', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
