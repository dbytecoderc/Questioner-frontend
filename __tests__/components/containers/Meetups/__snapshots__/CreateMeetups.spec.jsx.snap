// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders without errors 1`] = `
<div
  className="container"
>
  <h1>
    Create a meetup
  </h1>
  <Input
    className="form-control form-rounded"
    name="topic"
    onChange={[Function]}
    placeholder="Meetup topic..."
    type="text"
    value=""
  />
  <br />
  <Input
    className="form-control form-rounded"
    name="location"
    onChange={[Function]}
    placeholder="Meetup address..."
    type="text"
    value=""
  />
  <br />
  <Input
    className="form-control form-rounded"
    name="time"
    onChange={[Function]}
    placeholder="Meetup time"
    type="datetime-local"
    value=""
  />
  <br />
  <Button
    className="btn-dark submit--button"
    onClick={[Function]}
    type="submit"
    value="Post Meetup"
  />
</div>
`;
